src/wallet/wallet.cpp

```diff
@@ -4975,8 +5072,17 @@ void CWallet::GetFilteredNotes(
             SaplingOutPoint op = pair.first;
             SaplingNoteData nd = pair.second;
 
-            auto maybe_pt = SaplingNotePlaintext::decrypt(
-                wtx.vShieldedOutput[op.n].encCiphertext,
+            auto optDeserialized = SaplingNotePlaintext::attempt_sapling_enc_decryption_deserialization(wtx.vShieldedOutput[op.n].encCiphertext, nd.ivk, wtx.vShieldedOutput[op.n].ephemeralKey);
+
+            if (!optDeserialized) {
+                // The transaction would not have entered the wallet unless
+                // its plaintext had been successfully decrypted previously.
+                assert(false);
+            }
+            // We don't need to check the leadbyte here: if wtx exists in
+            // the wallet, it must have already passed the leadbyte check
+            auto maybe_pt = SaplingNotePlaintext::plaintext_checks_without_height(
```

Comment:
Wouldn't we have already performed called `plaintext_checks_without_height` when we originally decrypted the note and added it to our wallet? This check would then be redundant.